### ISSUE TITLE
[Desktop Excel Add-in] Error When Trying to Read a Named Range Referring to a Cell Range Using the OFFSET Formula

### ISSUE THREAD

**Issue by rbcxyz** [OfficeDev/office-js#4321]
This issue only occurs on the desktop version of Excel. Excel on the Web does not see this issue. 

I am developing an Excel Office Add-in using office-js and running into the following problem with named ranges.

Attempting to load a named range, or any properties of one, will throw an error if in the Name Manager, the Value section resolves to "{...}", which is the case when the "Refers To" of the named range is represented by an OFFSET formula. 

## Your Environment
* Platform: Windows Laptop
* Host: Excel
* Office version number: Version 2401 (Build 17231.20290 Click-to-Run)
* Operating System: Windows 10
 
## Expected behavior
Within my Excel Add-in code, I expect the following to load the values of my named range "MyRange" and print them to console:

```
const myRange = context.workbook.names.getItem("MyRange").getRange();
myRange.load("values");
await context.sync();
console.log(JSON.stringify(myRange));
```

## Current behavior
If MyRange refers to an OFFSET, such as for example: =OFFSET(Sheet1!$A$1,0,0,9,1), then the error "RichApi.Error: This operation is not permitted for the current object" is thrown upon calling context.sync() when the code in the above section is ran on the desktop version of excel. This will work without issue on the web version of excel though.

## Steps to reproduce
1. In Excel for desktop, define a new named range named "MyRange" where the "Refers to" value is an OFFSET formula referencing any range of cells within a sheet. For example '=OFFSET(Sheet1!$A$1,0,0,9,1)' to get the cell range A1:A9 in sheet Sheet1. The cell contents can be anything.
2. In the Add-ins menu on the Home page in Excel, open up the "Script Lab" Add-in.
3. Execute the following script through script lab:
```
$("#run").on("click", () => tryCatch(run));
 
async function run() {
  await Excel.run(async (context) => {
    const range = context.workbook.names.getItem("MyRange").getRange();
    range.load("values");
    await context.sync();
    console.log(JSON.stringify(range));
  });
}
 
/** Default helper for invoking an action and handling errors. */
async function tryCatch(callback) {
  try {
    await callback();
  } catch (error) {
    // Note: In a production add-in, you'd want to notify the user through your add-in's UI.
    console.error(error);
  }
}

```
3. Run the script and observe an error to be thrown on the line "await context.sync()"

## Context
I need to access the values/address within my named range objects

## Useful logs
Full error log from running this in script lab:
```
[ERROR]: Error:
{
    "stack": "RichApi.Error: This operation is not permitted for the current object.\n    at new n ([https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:263692)\n](https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:263692)/n)    at i.processRequestExecutorResponseMessage ([https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:328139)\n](https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:328139)/n)    at [https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:326202"](https://appsforoffice.microsoft.com/lib/1/hosted/excel-win32-16.01.js:25:326202%22),
    "message": "This operation is not permitted for the current object.",
    "name": "RichApi.Error",
    "code": "InvalidOperation",
    "traceMessages": [],
    "innerError": null,
    "debugInfo": {
        "code": "InvalidOperation",
        "message": "This operation is not permitted for the current object.",
        "errorLocation": "NamedItem.getRange",
        "statement": "var range = name.getRange();",
        "surroundingStatements": [
            "var workbook = context.workbook;",
            "var names = workbook.names;",
            "var name = names.getItem(...);",
            "// >>>>>",
            "var range = name.getRange();",
            "// <<<<<",
            "// Instantiate {range}",
            "range.load([\"values\"]);"
        ],
    },
    "httpStatusCode": 400
}
```

**Comment by donlvMSFT** [comment]
Hi @rbcxyz , 

Thanks for reporting this issue, I could reproduce the issue.
It has been put on our backlog <8899275> for internal track. We will keep track of this issue and update here to let you know if there are any updates.

**Comment by MitchellNZ** [comment]
+1
Experiencing this exact issue!

**Comment by rettoua** [comment]
+1
Same issue on Mac Desktop Excel

@shanshanzheng-dev is there any workaround ?

**Comment by penglongzhaochina** [comment]
Hi all,

We are working on this, will let you know once we have any progress.

**Comment by XuanZhouMSFT** [comment]
The issue is already fixed after version 16.0.17830.X for Windows and version 16.87 for MAC. Please have a try and let us know if you still have issues.
